scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -48,7 +75,16 @@ class BeamWriter(val beamMode: BeamMode) extends Writer {
               }
               rec(xs)
             }
-            //TODO: handle Force and ToIterable
+            case OptimizedWrite(pipe, ToWrite.ToIterable(opt)) =>
+              val pcoll = planner(opt).run(pipeline)
+              val tempLocation = pcoll.getPipeline.getOptions.getTempLocation
+              assert(tempLocation != null, "Temp location cannot be null when using toIterableExecution")
+
+              val outputPath = BeamWriter.addPaths(tempLocation, sourceCounter.getAndIncrement().toString)
+              new BeamFileIO(outputPath).write(pipeline, conf, pcoll)
```

Comment:
this looks like it is blocking here. We don't want that. We want to run this concurrently with all the other operations I think. Can you add a comment?

I think the loop should build a set of actions that need to run before the future that returns the run ID an execution counters.

Reply:
Added comment